docs/content/docs/operations/configuration.md

```diff
@@ -53,3 +53,12 @@ To learn more about metrics and logging configuration please refer to the dedica
 ## Operator Configuration Reference
 
 {{< generated/kubernetes_operator_config_configuration >}}
+
+## Job Specific Configuration Reference
+Job specific configuration can be configured under `spec.flinkConfiguration` and it will override flink configurations defined in `flink-conf.yaml`.
+
+- For application clusters, `spec.flinkConfiguration` will be located in `FlinkDeployment` CustomResource.
+- For session clusters, configuring `spec.flinkConfiguration` in parent `FlinkDeployment` will be applied to all session jobs within the session cluster.
+  - You can configure some additional job specific supplemental configuration through `spec.flinkConfiguration` in `FlinkSessionJob` CustomResource. 
+  Those session job level configurations will override the parent session cluster's Flink configuration. Please note only the following configurations are considered to be valid configurations.
+    - `kubernetes.operator.user.artifacts.http.header`
```

Comment:
I think it's much clear now 👍🏻. 
Some other thoughts come to my mind that, Maybe we could extend the `ConfigOption` to mark the config belong to which component (flink/operator). It's out of this PR, we could do it later if necessary.